lib/browser/methods/bots.rb

```diff
@@ -13,11 +13,23 @@ def self.detect_empty_ua?
     end
 
     def bot?
-      Browser::Bots.detect_empty_ua? && ua.strip == "" || BOTS.any? {|key, _| ua.include?(key) }
+      bot_with_empty_ua? || BOTS.any? {|key, _| ua.include?(key) }
+    end
+
+    def bot_name
+      return unless bot?
+      return "Generic Bot" if bot_with_empty_ua?
+      BOTS.find {|key, description| ua.include?(key) }.first
```

Comment:
Unused block argument - `description`. If it's necessary, use `_` or `_description` as an argument name to indicate that it won't be used.<br>Prefer `detect` over `find`.